tests/model/test_model.py

```diff
@@ -922,3 +923,32 @@ def test_handle_typing_event(self, mocker, model,
         model.handle_typing_event(event)
 
         assert model.controller.view.set_footer_text.called == called
+
+    @pytest.mark.parametrize('event, final_muted_streams, ', [
+        (
+            {'property': 'in_home_view',
+             'stream_id': 19,
+             'value': True},
+            {15}
+        ),
+        (
+            {'property': 'in_home_view',
+             'stream_id': 30,
+             'value': False},
+            {15, 19, 30}
+        )
+    ])
+    def test_update_subscription(self, model, mocker, event, stream_button,
+                                 final_muted_streams):
+        model.muted_streams = {15, 19}
+        model.controller.view.stream_id_to_button = {
+            19: stream_button,
+            30: stream_button
+        }
```

Comment:
This is dependent upon the parametrization; you could just set one entry based on the `event` values, and add a comment that we know the `stream_id`?

This perhaps raises the minor question of: can we get an event with a stream_id not in that dict? Should we handle that? Or at least comment this somewhere if it could be a future issue?

Reply:
Got it.
For now, this doesn't seem to be an issue, I'll add a comment as you suggested.